src/vos/vos_dtx.c

```diff
@@ -717,6 +730,469 @@ vos_dtx_abort_one(struct vos_container *cont, struct daos_tx_id *dti,
 	return rc;
 }
 
+struct vos_dtx_share {
+	/** Link into the daos_tx_handle::dth_shares */
+	d_list_t		vds_link;
+	/** The DTX record type, see enum vos_dtx_record_types. */
+	uint32_t		vds_type;
+	/** The record in the related tree in SCM. */
+	umem_id_t		vds_record;
```

Comment:
I see umem_id_t is used everywhere, actually we need only a 64bits offset, and that'll save us 64bits in each record. Let's make a cleanup patch to replace all these umem_id_t with 64bit offset later.

Reply:
OK, related optimization will be done next step.